plugins/GSdx/GSDevice.cpp

```diff
@@ -133,9 +133,9 @@ void GSDevice::Present(GSTexture* sTex, GSTexture* dTex, const GSVector4& dRect,
 
 GSTexture* GSDevice::FetchSurface(int type, int w, int h, bool msaa, int format)
 {
-	GSVector2i size(w, h);
+	const GSVector2i size(w, h);
 
-	for(list<GSTexture*>::iterator i = m_pool.begin(); i != m_pool.end(); ++i)
+	for(std::deque<GSTexture*>::const_iterator i = m_pool.cbegin(); i != m_pool.cend(); ++i)
```

Comment:
We can do a """worst case""" analysis: width * height * 4 * 32bit (RGBA) for each texture (no double precision right?), then make sure the total textures (pool + what is outside) doesn't exceed a given size.
When aging the pool or recycling, start by removing bigger textures, then fallback, until the target amount of free memory is reached.
We can trade a big framebuffer for a higher number of smaller textures in this way!
It wouldn't be perfect but better than nothing.